src/components/SessionTimeout/SessionTimeout.tsx

```diff
@@ -70,19 +70,27 @@ export class SessionTimeout extends React.Component<SessionTimeoutProps, {}> {
   };
 
   private textForLoginStrategy = () => {
+    const message =
+      this.props.timeOutCountDown <= 0
+        ? 'Your session has expired.'
+        : `Your session will expire in ${this.props.timeOutCountDown.toFixed()} seconds.`;
     return (
       <p className={'lead'}>
-        Your session will timeout in {this.props.timeOutCountDown.toFixed()} seconds.
+        {message}
         <br />
         Would you like to extend your session?
       </p>
     );
   };
 
   private textForOpenshiftStrategy = () => {
+    const message =
```

Comment:
Looks like some duplication of code here. Perhaps the `const message` can be moved out of the `textForLoginStrategy` and `textForOpenshiftStrategy` and put above these so it can be shared?

Reply:
OK, I changed things around in the hopes that his sire will find it aesthetically pleasing.